### ISSUE TITLE
Even if ``'babel'`` key is set to empty string, LaTeX output contains one `\addto\captions...` 

### ISSUE THREAD

**Issue by jfbu** [sphinx-doc/sphinx#3069]
On the other hand it does only `\renewcommand` on `\figurename` etc...

Perhaps https://github.com/sphinx-doc/sphinx/blob/master/sphinx/writers/latex.py#L434 is the culprit, I don't know why https://github.com/sphinx-doc/sphinx/blob/master/sphinx/writers/latex.py#L542 doesn't have the intended effect.

Typical output in latex with `'babel': ''`, in `latex_elements`

``` latex
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}
\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\renewcommand{\figurename}{Fig.\@ }
\renewcommand{\tablename}{Table }
\renewcommand{\literalblockname}{Listing }

\def\pageautorefname{page}
```

and with no change to `'babel'` one gets 

``` latex
% Set up styles of URL: it should be placed after hyperref.
\urlstyle{same}
\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\addto\captionsenglish{\renewcommand{\figurename}{Fig.\@ }}
\addto\captionsenglish{\renewcommand{\tablename}{Table }}
\addto\captionsenglish{\renewcommand{\literalblockname}{Listing }}

\addto\extrasenglish{\def\pageautorefname{page}}
```

This was tested with `language = None` and "howto" docclass.

**Comment by tk0miya** [comment]
Really? That's strange.
Can you reproduce that again?

**Comment by jfbu** [comment]
Yes, I can reproduce. Please find the conf.py and index.rst at https://gist.github.com/jfbu/3ef13b9c0848a1a0222c0f2bb9bf877a

With the empty `'babel': '',` in `latex_elements` compilation (on master branch) of  make latexpdf aborts with

```
! Undefined control sequence.
l.34 \addto
           \captionsenglish{\renewcommand{\contentsname}{Contents:}}
? 
```

illustrating the present issue. The produced tex file contains:

``` latex
\addto\captionsenglish{\renewcommand{\contentsname}{Contents:}}

\renewcommand{\figurename}{Fig.\@ }
\renewcommand{\tablename}{Table }
\renewcommand{\literalblockname}{Listing }

\def\pageautorefname{page}
```

This is tested on current master.

Secondly, if don't use empty `'babel'` key,  it produces this, which is wrong, because only `\refname` should have been used and not `\bibname` (sorry for mixing two issues but I already took screenshot and forgot I was working on this issue with empty `'babel'` key value).

![capture du 2016-10-20 15-00-57](https://cloud.githubusercontent.com/assets/2589111/19560803/28faa0b4-96d6-11e6-9929-ce7ffbd2cbbf.png)

**Comment by tk0miya** [comment]
Thanks, I can reproduce that!

**Comment by tk0miya** [comment]
I found the reason.

At master branch, the renewcommand macro for `\contentsname` is generated before recognize `latex_elements`.
So your empty `babel` key does not effect to the macro.
It's a bug of latex writer.

I will refactor the writer later.
Thanks, I've overlooked this ever...

**Comment by jfbu** [comment]
@tk0miya it seems there is even more serious problem with `contentsname`. Using:

``` python
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
extensions = []
templates_path = ['_templates']
source_suffix = '.rst'
master_doc = 'index'

# General information about the project.
project = 'testcontentsname'
copyright = '2016, jfbu'
author = 'jfbu'

version = '1'
release = '1'

language = 'de'

exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store']

# The name of the Pygments (syntax highlighting) style to use.
pygments_style = 'sphinx'

# If true, `todo` and `todoList` produce output, else they produce nothing.
todo_include_todos = False
# -- Options for LaTeX output ---------------------------------------------

latex_elements = {
#'babel' : '',
}

# Grouping the document tree into LaTeX files. List of tuples
# (source start file, target name, title,
#  author, documentclass [howto, manual, or own class]).
latex_documents = [
    (master_doc, 'a.tex', 'a Documentation',
     'jfbu', 'howto'),
]
```

I get in TeX file:

``` latex
\addto\captionsngerman{\renewcommand{\contentsname}{Contents:}}

\addto\captionsngerman{\renewcommand{\figurename}{Abb.\@ }}
\addto\captionsngerman{\renewcommand{\tablename}{Tab.\@ }}
\addto\captionsngerman{\renewcommand{\literalblockname}{Quellcode }}
```

i.e. the "table of contents" is not in German. Same with italian. Tested on master, stable, and... ok: 

It seems this is problem ever since 20d992cdf52c45a447947373e150b7d9f183084f which closed #1990 and is in 1.3.2 release. Maybe I open other issue ?

**Comment by jfbu** [comment]
@tk0miya oh sorry, it appears I got terribly confused due to use of a `:caption:` option which I had forgotten. Please disregard previous message. I will delete it.

**Comment by tk0miya** [comment]
Okay, never mind :-)

**Comment by tk0miya** [comment]
Fixed at #3124
